lib/elixir/lib/map.ex

```diff
@@ -691,8 +691,12 @@ defmodule Map do
       end
 
     case fun.(current) do
-      {get, update} -> {get, :maps.put(key, update, map)}
-      :pop          -> {current, :maps.remove(key, map)}
+      {get, update} ->
+        {get, :maps.put(key, update, map)}
+      :pop ->
+        {current, :maps.remove(key, map)}
+      other ->
+        raise "must return a two-element tuple or :pop, got: #{inspect(other)}"
```

Comment:
I think `RuntimeError` goes nicely with what we do in `Access`. Also this bad return can be result of some `if` expression inside that anonymous function.